File: 2015/python/day03/second.py
f = open("input.txt")
dict = {"0,0":1}	
i=ii=0
j=jj=0
while True:
	c = f.read(1)
	# print(c)
	if not c:
		break
	if c == '^':
		i+=1
	if c == 'v':
		i-=1
	if c == '>':
		j+=1
	if c == '<':
		j-=1

	key1 = str(i)+','+str(j)	
	if not key1 in dict:
		dict[key1] = 1
	else:
		dict[key1] +=1

	# print(dict)	

	c = f.read(1)
	# print(c)
	if not c:
		break
	if c == '^':
		ii+=1
	if c == 'v':
		ii-=1
	if c == '>':
		jj+=1
	if c == '<':
		jj-=1	


	key2 = str(ii)+','+str(jj)	
	if not key2 in dict:
		dict[key2] = 1
	else:
		dict[key2] +=1

	# print(dict)
		
print(len(dict))		

File: 2022/02/second.go
package main

import (
	"bufio"
	"log"
	"os"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatalf("can not open file")
	}

	defer file.Close()

	scanner := bufio.NewScanner(file)

	sum := 0

	// rock , paper , scissors
	eq := map[string]string{"A": "R", "B": "P", "C": "S"}

	beats := map[string]string{"A": "S", "B": "R", "C": "P"}

	beated := map[string]string{"A": "P", "B": "S", "C": "R"}

	score := map[string]int{"R": 1, "P": 2, "S": 3}

	for scanner.Scan() {
		v := strings.Fields(scanner.Text())

		if v[1] == "X" {
			sum += score[beats[v[0]]]
		} else if v[1] == "Y" {
			sum += score[eq[v[0]]] + 3
		} else {
			sum += score[beated[v[0]]] + 6
		}

	}

	log.Println(sum)
}
